keps/sig-windows/windows-runtimeclass-support.md

```diff
@@ -297,53 +300,98 @@ To make this easier to consume in the Kubelet and APIs, it will be updated in mu
 
-The proposal is to do the same for [PullImageRequest](https://github.com/kubernetes/cri-api/blob/24ae4d4e8b036b885ee1f4930ec2b173eabb28e7/pkg/apis/runtime/v1alpha2/api.proto#L1081). With the additions, it would be:
+```
+“kubernetes.io/runtimehandler”: “<corresponding values>”
+ ```
 
+We could potentially also add the kubernetes specification annotations for consideration of the runtime:
+```
+“kubernetes.io/arch”: “amd64”
+“kubernetes.io/os”: ”linux”
 ```
-message PullImageRequest {
-    // Spec of the image.
-    ImageSpec image = 1;
-    // Authentication configuration for pulling the image.
-    AuthConfig auth = 2;
-    // Config of the PodSandbox, which is used to pull image in PodSandbox context.
-    PodSandboxConfig sandbox_config = 3;
-    // Named runtime configuration to use for this PodSandbox.
-    // This should match a runtime_handler used in RunPodSandboxRequest
-    // and is subject to the same semantics.
-    string runtime_handler = 4;
+
+Note that these are currently derived from from GOARCH and GOOS at runtime, which does not reflect the image which needs to be pulled but instead corresponds to the system on which kubelet is running. The runtime handler specified in the annotation will provide more accurate indication of the user intent to the runtime.
+
+##### ImageService Interface changes to include ImageSpec
+
+There are calls in which the ImageSpec are returned back from the runtime to Kubelet:
+
+```
+type ImageService interface {
+	// PullImage pulls an image from the network to local storage using the supplied
+	// secrets if necessary. It returns a reference (digest or ID) to the pulled image.
+	PullImage(image ImageSpec, pullSecrets []v1.Secret, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, error)
+	// GetImageRef gets the reference (digest or ID) of the image which has already been in
+	// the local storage. It returns ("", nil) if the image isn't in the local storage.
+	GetImageRef(image ImageSpec) (string, error)
+	// Gets all images currently on the machine.
+	ListImages() ([]Image, error)
+	// Removes the specified image.
+	RemoveImage(image ImageSpec) error
+	// Returns Image statistics.
+	ImageStats() (*ImageStats, error)
+}
+```
+
+The proposal is to change the calls which don't return ImageSpec like the ListImage to return the ImageSpec:
+```
+type ImageService interface {
+    // PullImage pulls an image from the network to local storage using the supplied
+    // secrets if necessary. It returns a reference (digest or ID) to the pulled image.
+    PullImage(image ImageSpec, pullSecrets []v1.Secret, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, error)
+    // GetImageRef gets the reference (digest or ID) of the image which has already been in
+    // the local storage. It returns ("", nil) if the image isn't in the local storage.
+    GetImageRef(image ImageSpec) (string, error)
+    // Gets all images currently on the machine.
+    ListImages() ([]ImageSpec, error)
```

Comment:
`ListImages` should continue return Images, we just need to add ImageSpec information into `Image`.

Reply:
ok - can we work out the specifics of that during phase 2 PR? We don't have the background on why `Image` and `ImageSpec` are different

Reply:
Thanks you @Random-Liu . I have modified the struct Image to include the ImageSpec in the updated PR. Please have a look - @Random-Liu @PatrickLang

Reply:
oh ok - that makes sense

Reply:
/lgtm
/hold
(remove once @Random-Liu ok with final update)